packages/okta-adapter/src/reference_mapping.ts

```diff
@@ -19,6 +19,19 @@ import { GetLookupNameFunc } from '@salto-io/adapter-utils'
 import { APPLICATION_TYPE_NAME, GROUP_TYPE_NAME, USER_TYPE_NAME, IDENTITY_PROVIDER_TYPE_NAME, USERTYPE_TYPE_NAME, FEATURE_TYPE_NAME, POLICY_TYPE_NAME, NETWORK_ZONE_TYPE_NAME, APP_USER_TYPE_NAME, ROLE_TYPE_NAME } from './constants'
 
 
+export class OktaFieldReferenceResolver extends referenceUtils.FieldReferenceResolver<never> {
+  constructor(def: referenceUtils.FieldReferenceDefinition<never>) {
+    super({ src: def.src })
+    this.serializationStrategy = referenceUtils.ReferenceSerializationStrategyLookup[
+      def.serializationStrategy ?? def.serializationStrategy ?? 'fullValue'
+    ]
+    this.target = def.target
+      ? { ...def.target, lookup: this.serializationStrategy.lookup }
+      : undefined
+    this.sourceTransformation = referenceUtils.ReferenceSourceTransformationLookup[def.sourceTransformation ?? 'asString']
```

Comment:
the comment from https://github.com/salto-io/salto/pull/3627#discussion_r1040053423 became outdated but it still applies - I think this can just be
```suggestion
super({ src: def.src, sourceTransformation: def.sourceTransformation ?? 'asString'  })
```